lib/mixlib/install/options.rb

```diff
@@ -110,6 +110,10 @@ def latest_version?
         product_version.to_sym == :latest
       end
 
+      def partial_version?
+        !Mixlib::Versioning.parse(product_version) && !latest_version?
```

Comment:
I think this approach is good as it yields the actual parsing down to mixlib-versioning...which means we don't maintain this parsing logic in two separate spots. You might want to flip your AND check and do `!latest_version?` first as that would save parsing the version if it is the `:latest`.